File: replaceDelimiters.js
const fs = require('fs');
const path = require('path');

function replaceEquationDelimiters(text) {
    const pattern = /<div class="equation">(.*?)<\/div>/gs;
    const replacement = (match) => match.replace(/\$\$(.*?)\$\$/gs, ' \\\\\[ $1 \\\\\]');
    return text.replace(pattern, replacement);
}

function replaceAllDelimiters(text) {
    const pattern = /\$\$/g;
    const replacement = '$';
    return text.replace(pattern, replacement);
}

const folderPath = 'contents';

fs.readdir(folderPath, (err, files) => {
    if (err) {
        console.error(err);
        return;
    }

    files.forEach((fileName) => {
        if (fileName.endsWith('.md')) {
            const filePath = path.join(folderPath, fileName);

            fs.readFile(filePath, 'utf-8', (err, data) => {
                if (err) {
                    console.error(err);
                    return;
                }

                let content = data;
                content = replaceEquationDelimiters(content);
                content = replaceAllDelimiters(content);

                fs.writeFile(filePath, content, 'utf-8', (err) => {
                    if (err) {
                        console.error(err);
                    }
                });
            });
        }
    });
});

File: replaceDelimiters.py
import re
import os

def replace_equation_delimiters(text):
    pattern = r'<div class="equation">(.*?)<\/div>'
    replacement = lambda match: re.sub(r'\$(.*?)\$', r'$$\1$$', match.group(0))
    return re.sub(pattern, replacement, text, flags=re.DOTALL)

def replace_all_delimiters(text):
    pattern = r'\$\$'
    replacement = r'$'
    return re.sub(pattern, replacement, text)

# Path to the contents folder
folder_path = 'contents'

# Iterate over files in the folder
for file_name in os.listdir(folder_path):
    if file_name.endswith('.md'):
        file_path = os.path.join(folder_path, file_name)
        
        # Read the content from the file
        with open(file_path, 'rt', encoding='utf-8') as file:
            content = file.read()
        
        # Replace equation delimiters
        updated_content = replace_all_delimiters(content)
        updated_content1 = replace_equation_delimiters(updated_content)
        
        # Write the updated content back to the file
        with open(file_path, 'w', encoding="utf-8") as file:
            file.write(updated_content1)